### PR TITLE
extra_repr for DropBlock layer

### PR DESCRIPTION
**Pull request by guillaumefrd**
## Motivation

Add `extra_repr` in DropBlock layer to get details in the model print.

```
print(model)

# before
(dropblock1): DropBlock()

# now
(dropblock1): DropBlock(drop_prob=0.1, block_size=7, warmup_iters=2000)
```

## Modification

Add `extra_repr` in DropBlock layer

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/models/plugins/dropblock.py [hunk 1]
```diff
@@ -79,3 +79,8 @@ def _compute_gamma(self, feat_size):
         factor = (1.0 if self.iter_cnt > self.warmup_iters else self.iter_cnt /
                   self.warmup_iters)
         return gamma * factor
+
+    def extra_repr(self):
+        s = ('drop_prob={drop_prob}, block_size={block_size}, '
```
Comment (hhaAndroid) [inline mmdet/models/plugins/dropblock.py:84]: drop_prob={self.drop_prob}  ...

Comment (guillaumefrd) [reply mmdet/models/plugins/dropblock.py]: @hhaAndroid I initially went for the same coding style than `extra_repr` of Conv in pytorch repo ([code here](https://github.com/pytorch/pytorch/blob/7e7be526c9d9179f35084e9cca5b5c5ad5172100/torch/nn/modules/conv.py#L150)). 
It is now replaced with your requested change.

File: mmdet/models/plugins/dropblock.py [hunk 2]
```diff
@@ -79,3 +79,7 @@ def _compute_gamma(self, feat_size):
         factor = (1.0 if self.iter_cnt > self.warmup_iters else self.iter_cnt /
                   self.warmup_iters)
         return gamma * factor
+
+    def extra_repr(self):
+        return (f'drop_prob={self.drop_prob}, block_size={self.block_size}, '
```
Comment (hhaAndroid) [inline mmdet/models/plugins/dropblock.py:84]: Why the return value is not str？

Comment (hhaAndroid) [inline mmdet/models/plugins/dropblock.py:84]: This type of return (f'xxx') is not recommended, it is recommended to return 'xxx'

**Review by ZwwWayne — APPROVED** [review]

**Comment by OpenMMLab-Coodinator** [comment]
Hi @guillaumefrd ！First of all, we want to express our gratitude for your significant PR in this project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you are Chinese or have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**